### ISSUE TITLE
FilePicker with ref error

### ISSUE THREAD

**Issue by BrentHuang** [flet-dev/flet#3239]
**Description**

**Code example to reproduce the issue:**

```python
import flet as ft

file_picker = ft.Ref[ft.FilePicker]()


def main(page: ft.Page):

    def pick_files_result(e: ft.FilePickerResultEvent):
        selected_files.value = (", ".join(map(lambda f: f.name, e.files)) if e.files else "Cancelled!")
        selected_files.update()

    pick_files_dialog = ft.FilePicker(on_result=pick_files_result, ref=file_picker)
    selected_files = ft.Text()

    page.overlay.append(pick_files_dialog)

    page.add(
        ft.Row([
            ft.ElevatedButton(
                "Pick files",
                icon=ft.icons.UPLOAD_FILE,
                on_click=lambda _: pick_files_dialog.pick_files(allow_multiple=True),
            ),
            selected_files,
        ]))


ft.app(target=main)
```

**Describe the results you received:**

run with 'flet run ./demo.py', the result is:

![image](https://github.com/flet-dev/flet/assets/2862064/1c7a6d96-8359-45b1-83f1-7277a3c6384f)

the callback pick_files_result not entered.

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Flet version (`pip show flet`)**:

```
0.22.0
```

**Give your `requirements.txt` file (don't `pip freeze`, instead give direct packages):**

```
no
```

**Operating system**:

Linux Mint 21.3 Virginia \n \l

**Additional environment details:**

**Comment by BrentHuang** [comment]
on windows, it runs ok.

**Comment by ndonkoHenri** [comment]
What do you mean by "ref error" in the issue title?

**Comment by BrentHuang** [comment]
> What do you mean by "ref error" in the issue title?

![image](https://github.com/flet-dev/flet/assets/2862064/e3412588-50e6-4d26-b47a-b8e4a14e0f5b)

**Comment by ndonkoHenri** [comment]
Yep, I mean, what's the error with them?

**Comment by BrentHuang** [comment]
the rusult is the loop picture: 
![image](https://github.com/flet-dev/flet/assets/2862064/38df2442-c7ac-4829-b07c-d0479854494f)

not ‘pick_files_result’ callback

**Comment by BrentHuang** [comment]
if you have a linux environment, you can try my code and click the button. This problem is inevitable.

**Comment by lekshmanmj** [comment]
I tried your code in web mode... I didn't face issue.

I've a question... Were you able to run app in desktop mode using `flet run` command ?

**Comment by BrentHuang** [comment]
yes, I run app in desktop mode using `flet run` command. my problem is desktop app on LinuxMInt.

> I tried your code in web mode... I didn't face issue.我在网络模式下尝试了你的代码...我没有遇到问题。
> 
> I've a question... Are you able to run app in desktop mode using `flet run` command ?我有一个问题...您可以使用 `flet run` 命令在桌面模式下运行应用程序吗？
